editor/py/serve.py

```diff
@@ -90,6 +97,34 @@ def upload_dataset(file_name):
   return 'ok'
 
 
+@app.route('/dataset/<file_name>/new', methods=['POST'])
+def new_dataset(file_name):
+  """Creates a new dataset in the db/ directory."""
+  path = f'db/{flask.g.user}/{file_name}.pbtxt'
+  if os.path.isfile(path):
+    return 'file exists'
```

Comment:
Yes, perhaps. I didn't write any error handling. What do you recommend?

Reply:
Since I still don't understand the purpose of this endpoint, I can't comment on its error semantics.

So far, we have not been propagating I/O errors from the backend. A first step would be to show a dialog so the person can know their change didn't take or their data have been corrupted. The natural way to express this on the wire would be some 4XX status code.

It's still OK with me to defer this stuff. Steven is working on a DB anyway, so both the backend and its variety of errors will be different soon.

Reply:
Changed to return a 404 & added basic JS error handling to provide an `alert` that the dataset already exists